go.mod

```diff
@@ -1,6 +1,8 @@
 module github.com/sylabs/singularity/v4
 
-go 1.20
+go 1.21
+
+toolchain go1.21.8
```

Comment:
If I understand correctly, the `toolchain` line will cause an installed version of Go between 1.21.0 and 1.21.7 to download and use 1.21.8 via automatic toolchain switching. Is that the intent here? And if so, will we plan to bump this each time the Go team releases a patch release?

Haven't seen use of this in the wild yet, so not sure what the best practice is!

Reply:
> If I understand correctly, the `toolchain` line will cause an installed version of Go between 1.21.0 and 1.21.7 to download and use 1.21.8 via automatic toolchain switching. Is that the intent here? 

That's my understanding of the behaviour, yes.

I included it here, because on dependabot updates our go.mod check is complaining about it missing... see e.g. the dependabot PRs open for the CNI plugins:

<img width="463" alt="image" src="https://github.com/sylabs/singularity/assets/4522799/eb1269d6-3419-4a05-883a-eb9a101bd509">

> And if so, will we plan to bump this each time the Go team releases a patch release?
> Haven't seen use of this in the wild yet, so not sure what the best practice is!

I'm a bit unclear on this tbh. My gut feeling is that for projects like ours, it is useful to inform a minimum Go toolchain version to avoid a relevant security issue. Therefore I'd anticipate we'd bump it if there's some security issue in a Go release that's relevant for Singularity?

We could drop the toolchain line for now... and then work around anything that comes up on the check go.mod task?

Reply:
After some discussion... will drop the toolchain line at this time, to avoid issues with distros etc. where fetching a toolchain would be frowned upon.